### PR TITLE
Update maven publishing

### PR DESCRIPTION
**Pull request by martinbonnin**
* use the Android software components available with AGP 3.6
* do not create publications if the java-gradle-plugin or org.jetbrains.kotlin.multiplatform did it for us
* properly configure the pom metadata
* remove the bintrayMarker repository
* upload everything to the merged `apollo` package
* since the publications do not overlap anymore, we can call "publishAllPublications"

### CODE REVIEW

**Comment by tasomaniac** [comment]
@martinbonnin can you check if this is ok with these changes? https://github.com/apollographql/apollo-android/issues/2140

**Comment by martinbonnin** [comment]
Yes, that should fix it.

File: build.gradle.kts [hunk 1]
```diff
@@ -209,43 +208,43 @@ fun Project.configurePublishing() {
...
-            artifactId = findProperty("POM_ARTIFACT_ID") as String?
+            pom {
+              artifactId = findProperty("POM_ARTIFACT_ID") as String?
+            }
           }
         }
       }
+
+      withType<MavenPublication>() {
+        setDefaultPomFields()
```
Comment (martinbonnin) [inline build.gradle.kts:246]: This was missing and preventing the license url to be present in the pom files

File: apollo-normalized-cache-sqlite/gradle.properties [hunk 2]
```diff
@@ -1,4 +1,4 @@
-POM_ARTIFACT_ID=apollo-sqlite-cache
+POM_ARTIFACT_ID=apollo-normalized-cache-sqlite
```
Comment (tasomaniac) [inline apollo-normalized-cache-sqlite/gradle.properties:1]: Ops

**Pull request merged**